### PR TITLE
EVG-3270 Noop setup job if host is already set up

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

**Comment by bsamek** [comment]
This is nearly all code reorganization. The only change is this `if` block https://github.com/evergreen-ci/evergreen/pull/1206/files#diff-b68d7c29013acf30e83cf282cd0f00feR434.

File: units/provisioning_agent_deploy.go [hunk 1]
```diff
@@ -104,3 +112,229 @@ func (j *agentDeployJob) Run(ctx context.Context) {
...
+		m["last_communication_at_time"] = sinceLCT.String()
+	}
+
+	return m
+}
+
+// Start an agent on the host specified.  First runs any necessary
+// preparation on the remote machine, then kicks off the agent process on the
+// machine. Returns an error if any step along the way fails.
+func StartAgentOnHost(ctx context.Context, settings *evergreen.Settings, hostObj host.Host) error {
```
Comment (tychoish) [inline units/provisioning_agent_deploy.go]: why is this public? why is this in the units package?

File: units/provisioning_setup_host.go [hunk 2]
```diff
@@ -75,5 +93,749 @@ func (j *setupHostJob) Run(ctx context.Context) {
...
+	script, err := exp.ExpandString(s)
+	if err != nil {
+		return "", errors.Wrap(err, "expansions error")
+	}
+	return script, err
+}
+
+// Provision the host, and update the database accordingly.
+func provisionHost(ctx context.Context, h *host.Host, settings *evergreen.Settings) error {
+	if h.Status == evergreen.HostRunning {
```
Comment (tychoish) [inline units/provisioning_setup_host.go]: put this in the Run()

**Comment by bsamek** [comment]
* made all public functions private methods on the job objects
* moved noop to top level of job

**Pull request merged**